src/prefect_server/api/flows.py

```diff
@@ -261,37 +266,56 @@ async def create_flow(
         description=description,
         schedule=serialized_flow.get("schedule"),
         is_schedule_active=False,
-        tasks=[
-            models.Task(
-                id=t.id,
-                tenant_id=tenant_id,
-                name=t.name,
-                slug=t.slug,
-                type=t.type,
-                max_retries=t.max_retries,
-                tags=t.tags,
-                retry_delay=t.retry_delay,
-                trigger=t.trigger,
-                mapped=t.mapped,
-                auto_generated=t.auto_generated,
-                cache_key=t.cache_key,
-                is_reference_task=t.is_reference_task,
-                is_root_task=t.is_root_task,
-                is_terminal_task=t.is_terminal_task,
+        tasks=[],
+        edges=[],
+    ).insert()
+
+    try:
+        batch_insertion_size = 2500
+
+        for tasks_chunk in get_chunk(flow.tasks, batch_insertion_size):
+            await models.Task.insert_many(
+                [
+                    models.Task(
+                        id=t.id,
+                        flow_id=flow_id,
+                        tenant_id=tenant_id,
+                        name=t.name,
+                        slug=t.slug,
+                        type=t.type,
+                        max_retries=t.max_retries,
+                        tags=t.tags,
+                        retry_delay=t.retry_delay,
+                        trigger=t.trigger,
+                        mapped=t.mapped,
+                        auto_generated=t.auto_generated,
+                        cache_key=t.cache_key,
+                        is_reference_task=t.is_reference_task,
+                        is_root_task=t.is_root_task,
+                        is_terminal_task=t.is_terminal_task,
+                    )
+                    for t in tasks_chunk
+                ]
             )
-            for t in flow.tasks
-        ],
-        edges=[
-            models.Edge(
-                tenant_id=tenant_id,
-                upstream_task_id=task_lookup[e.upstream_task].id,
-                downstream_task_id=task_lookup[e.downstream_task].id,
-                key=e.key,
-                mapped=e.mapped,
+
+        for edges_chunk in get_chunk(flow.edges, batch_insertion_size):
+            await models.Edge.insert_many(
+                [
+                    models.Edge(
+                        tenant_id=tenant_id,
+                        flow_id=flow_id,
+                        upstream_task_id=task_lookup[e.upstream_task].id,
+                        downstream_task_id=task_lookup[e.downstream_task].id,
+                        key=e.key,
+                        mapped=e.mapped,
+                    )
+                    for e in edges_chunk
+                ]
             )
-            for e in flow.edges
-        ],
-    ).insert()
+
+    except Exception as exception:
+        await delete_flow(flow_id=flow_id)
+        raise exception
```

Comment:
We might want to reraise this as an `APIError` although I'm not sure where we want to do this generally. Perhaps it belongs in the graphql layer. cc @cicdw

Reply:
Hmmmm yea, at a minimum I'd vote to log this error and raise a custom exception "Failed to register Flow" instead of raising the generic exception -- an APIError would work too